neo/Ledger/Blockchain.cs

```diff
@@ -364,14 +380,24 @@ private RelayResultReason OnNewTransaction(Transaction transaction)
         private void OnPersistCompleted(Block block)
         {
             block_cache.Remove(block.Hash);
-            foreach (Transaction tx in block.Transactions)
-                mem_pool.TryRemove(tx.Hash, out _);
-            foreach (Transaction tx in mem_pool
+
+            foreach (var tx in block.Transactions)
+            {
+                if (!mem_pool_free.TryRemove(tx.Hash, out _))
```

Comment:
Maybe a verification can avoid Try to Remove from free if tx has fees.
something like tx.fees, right?

Reply:
You are right :)

Reply:
Thanks, myy friend :dagger: